manager/src/main/java/com/redhat/service/smartevents/manager/workers/quartz/WorkJob.java

```diff
@@ -41,6 +52,12 @@ private Worker<?> findWorker(Work work) {
             return bridgeWorker;
         } else if (Processor.class.getName().equals(work.getType())) {
             return processorWorker;
+        } else if (LEGACY_BRIDGE_WORKER_CLASSNAME.equals(work.getType())) {
+            // Existing Work may have been created on an older version
+            return legacyBridgeWorker;
```

Comment:
@manstis Do we need the extra `Legacy{Bridge|Processor}Worker` classes? IIUC we can avoid that with a simple switch statement?

```
switch(work.getType()) {
  case Bridge.class.getName():
  case LEGACY_BRIDGE_CLASSNAME:
      return bridgeWorker;
```

Or do I mis-understand the intent here?

Reply:
You're not missing anything. Refactored.